test/cql-pytest/test_using_timestamp.py

```diff
@@ -90,3 +90,79 @@ def test_key_writetime(cql, table1):
         cql.execute(f'SELECT writetime(k) FROM {table1}')
     with pytest.raises(InvalidRequest, match='PRIMARY KEY part k'):
         cql.execute(f'SELECT ttl(k) FROM {table1}')
+
+# Reproducer for https://github.com/scylladb/scylladb/issues/14182
+def test_rewrite_using_same_timestamp(cql, table1):
+    table = table1
+    k = unique_key_int()
+    ts = 1000
+    ttl1 = 1
+    ttl2 = 1
+    values = [[1, 1], [1, 2], [2, 1]]
+    errors = []
+    for i in range(3):
+        v1, v2 = values[i]
+
+        # sync with wall-clock on exact second so that expiration won't cross the whole-second boundary
+        t1 = time.time()
+        while t1 - int(t1) >= 0.9:
+            time.sleep(1 - (t1 - int(t1)))
+            t1 = time.time()
+
+        cql.execute(f"INSERT INTO {table} (k, v) VALUES ({k}, {v1}) USING TIMESTAMP {ts} AND TTL {ttl1}")
+        # rewriting right away, so that both cells will carry the same expiry and ttl should return the larger value
+        rewrite = f"INSERT INTO {table} (k, v) VALUES ({k}, {v2}) USING TIMESTAMP {ts} AND TTL {ttl2}"
+        cql.execute(rewrite)
+        select = f"SELECT k, v FROM {table} WHERE k = {k}"
+        res = list(cql.execute(select))
+        assert len(res) == 1, f"Expected a single row, but got {res}"
```

Comment:
General comment about pytest: `assert len(res) == 1` is enough, you don't need to invent a different failure message. That's one of the most beautiful features of pytest: It instruments "assert()" calls to give you a lot of extra information when it fails. In this case it will tell you exactly which assert call fails (the entire call tree), and tell you what was `len(res)` and even what was `res` when this failed. It's beautiful.

Reply:
yeah, I'm too familiar with pytest assertion failure messages.
The only problem with them is that they abbreviate long values, while the hand-written error message provides the full `res` list.

Reply:
Almost two years ago, pytest added the "-vv" option (the v option twice) that will avoid this truncation of long values.
Note that even when long lists are involved, pytest many times managest to print the right part. For example, if you do "assert l1 == l2", it doesn't prints specifically which part of l1 and l2 differ, not a hundred items which are equal. In our C++ tests and dtest we jumped through hoops to add these features, but pytest gives it to use for free - you just need to use "assert". This pytest feature is called "assertion rewriting" and is perhaps my favorite part of pytest.

Check out:
https://www.pythoninsight.com/2018/01/assertion-rewriting-in-pytest-part-1/
https://www.pythoninsight.com/2018/02/assertion-rewriting-in-pytest-part-4-the-implementation/

Reply:
got rid of the customized error messages in v4